classes/image_types_ota.bbclass

```diff
@@ -108,24 +69,13 @@ IMAGE_CMD_ota () {
 	echo "{\"${ostree_target_hash}\":\"${GARAGE_TARGET_NAME}-${target_version}\"}" > ${OTA_SYSROOT}/ostree/deploy/${OSTREE_OSNAME}/var/sota/import/installed_versions
 }
 
+EXTRA_IMAGECMD_ota-ext4 = "-O ^64bit,^metadata_csum -L otaroot -i 4096"
 IMAGE_TYPEDEP_ota-ext4 = "ota"
-do_image_ota_ext4[depends] = "e2fsprogs-native:do_populate_sysroot"
+IMAGE_ROOTFS_task-image-ota-ext4 = "${OTA_SYSROOT}"
 IMAGE_CMD_ota-ext4 () {
-	# Calculate image size
-	OTA_ROOTFS_SIZE=$(calculate_size `du -ks ${OTA_SYSROOT} | cut -f 1`  "${IMAGE_OVERHEAD_FACTOR}" "${IMAGE_ROOTFS_SIZE}" "${IMAGE_ROOTFS_MAXSIZE}" `expr ${IMAGE_ROOTFS_EXTRA_SPACE}` "${IMAGE_ROOTFS_ALIGNMENT}")
-
-	if [ ${OTA_ROOTFS_SIZE} -lt 0 ]; then
-		bbfatal "create_ota failed to calculate OTA rootfs size!"
-	fi
-
-	eval local COUNT=\"0\"
-	eval local MIN_COUNT=\"60\"
-	if [ ${OTA_ROOTFS_SIZE} -lt ${MIN_COUNT} ]; then
-		eval COUNT=\"${MIN_COUNT}\"
-	fi
-
-	dd if=/dev/zero of=${IMGDEPLOYDIR}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.ota-ext4 seek=${OTA_ROOTFS_SIZE} count=${COUNT} bs=1024
-	mkfs.ext4 -O ^64bit ${IMGDEPLOYDIR}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.ota-ext4 -L otaroot -d ${OTA_SYSROOT}
+	ln -sf ${STAGING_DIR_NATIVE}${base_sbindir_native}/mkfs.ext4 ${STAGING_DIR_NATIVE}${base_sbindir_native}/mkfs.ota-ext4
+	ln -sf ${STAGING_DIR_NATIVE}${base_sbindir_native}/fsck.ext4 ${STAGING_DIR_NATIVE}${base_sbindir_native}/fsck.ota-ext4
+	oe_mkext234fs ota-ext4 ${EXTRA_IMAGECMD}
```

Comment:
This looks cool, but I'm failing to see how the variables we used before to control the size get used by this function. I'm a bit worried about unexpected results here.

Reply:
@patrickvacek 

Previously we were using:
```
OTA_ROOTFS_SIZE=$(calculate_size `du -ks ${OTA_SYSROOT} | cut -f 1`  "${IMAGE_OVERHEAD_FACTOR}" "${IMAGE_ROOTFS_SIZE}" "${IMAGE_ROOTFS_MAXSIZE}" `expr ${IMAGE_ROOTFS_EXTRA_SPACE}` "${IMAGE_ROOTFS_ALIGNMENT}")
```

the logic is basically same with OE's function get_rootfs_size, which is being called for every image task in prefuncs, if you could add some print in openembedded-core/meta/classes/image.bbclass, for instance:
```
diff --git a/meta/classes/image.bbclass b/meta/classes/image.bbclass
index adc50c99bc..06cac9341a 100644
--- a/meta/classes/image.bbclass
+++ b/meta/classes/image.bbclass
@@ -518,6 +518,7 @@ def get_rootfs_size(d):
     initramfs_fstypes = d.getVar('INITRAMFS_FSTYPES') or ''
     initramfs_maxsize = d.getVar('INITRAMFS_MAXSIZE')
 
+    bb.warn("calling du -ks to: %s" % d.getVar('IMAGE_ROOTFS'))
     output = subprocess.check_output(['du', '-ks',
                                       d.getVar('IMAGE_ROOTFS')])
     size_kb = int(output.split()[0])
```

you will see how it's getting called for do_image_ota_ext4

Reply:
@aodukha is this something you can verify? I'm a bit too busy at the moment to dig into the details of how the image sizes are calculated.

Reply:
in order to verify it I will builds rapspi 3 & quemu (should not be affected at all) and flash pi3-4. but it will take some time. 
IMHO this should not affect our builds. but make sense integrate into master firstly and check differences between returned values from get_rootfs_size and current values dunfell/warrior/etc. It is possible (but unlikely) that somebody depends on it

Reply:
Yeah, I think just building a couple images and making sure the sizes look good is worth checking. I think this PR looks good and I'd prefer to use standardized code instead of this older, cruftier custom code, but I need some more confidence that it won't do something unexpected.

Since master CI is ugly, it shouldn't be hard to try this change out on dunfell instead of master if that helps.

Reply:
at least for qemu build existing version and PR calls dd with 209767. hm.. similar numbers (checked twice, have to check rasp version too)
make sense run it on CI. it seems to have identical functionality to existing code and a bit smaller code

Reply:
Cool, if it's similar, then it's probably good to go. I just wanted to make sure it was double-checked.

Reply:
Approved. there are no 
flashed and run rasp3 and quemu with this. Using local build for this.
I run CI for dunfell I https://main.gitlab.in.here.com/olp/edge/ota/connect/client/meta-updater/-/pipelines/1706123

Reply:
This doesn't seem to work very well as it produces ext2 instead of ext4 now, here are the fixes for master and dunfell:
https://github.com/advancedtelematic/meta-updater/pull/790
https://github.com/advancedtelematic/meta-updater/pull/791

Reply:
> This doesn't seem to work very well as it produces ext2 instead of ext4 now, here are the fixes for master and dunfell:
> #790
> #791

Yes, my bad, "-t ext4" is mandatory, but I have a concern about these pull request, had some comments on: https://github.com/advancedtelematic/meta-updater/pull/790/commits/64c24452a850001223df1054bfcbb1a9010732ac

Reply:
:( will run CI asap to check the fix